conjure-test/src/test/objects.rs

```diff
@@ -128,6 +128,16 @@ fn unions() {
     test_ser(&unknown_value, unknown_json);
 }
 
+#[test]
+fn typeless_unions() {
```

Comment:
worth testing the case of deserializing empty optional? (one of the edge cases described in  https://github.com/palantir/conjure/issues/111)